### PR TITLE
pd/tls: add upgrade tls description

### PR DESCRIPTION
**Pull request by HuSharp**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.6 (TiDB Operator 1.6 versions)
- [x] v1.5 (TiDB Operator 1.5 versions)
- [x] v1.4 (TiDB Operator 1.4 versions)
- [ ] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):  https://github.com/pingcap/tidb-operator/pull/5708

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: zh/enable-tls-between-components.md [hunk 1]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
     > **注意：**
     >
     > 在集群创建后，不能修改此字段，否则将导致集群升级失败，此时需要删除已有集群，并重新创建。
+    > 
+    > 集群创建后再开启 tls，在无法重建的情况下，可尝试以下方式（**请谨慎操作**）：
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:25]:
```suggestion
    > 集群创建后再开启 TLS，在无法重建的情况下，可尝试以下方式（**请谨慎操作**）：
```

File: zh/enable-tls-between-components.md [hunk 2]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    > 
+    > 1. 在存在多个 PD 节点的情况下，缩容 PD 至 1 个。（否则可以跳过此步骤）
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:27]:
```suggestion
    > 1. 如果存在多个 PD 节点，缩容 PD 至 1 个节点。
```

File: zh/enable-tls-between-components.md [hunk 3]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    > 2. 下载 etcdctl。参考 [etcdctl 安装指南](https://etcd.io/docs/v3.4/install/)，etcdctl 位于文件夹目录下。
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:28]:
```suggestion
    > 2. 参考 [etcdctl 安装指南](https://etcd.io/docs/v3.4/install/) 下载安装 `etcdctl`。
```

File: zh/enable-tls-between-components.md [hunk 4]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    > 3. 设置 `.spec.tlsCluster.enabled` 属性为 `true`；
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:29]:
```suggestion
    > 3. 将 `.spec.tlsCluster.enabled` 更新为 `true` 并等待 PD Pod 重启。
```

File: zh/enable-tls-between-components.md [hunk 5]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    > 4. 查看 etcd member，可见 peerURLs 此时为 http：
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:30]: 这里是打算让用户把 `etcdctl` 放到 Pod 里面，还是把证书取出来并且 port-forward？

File: zh/enable-tls-between-components.md [hunk 6]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    >    >
+    >    > ```bash
+    >    > etcdctl --endpoints https://127.0.0.1:2379 --cert-file ./pd-tls/tls.crt --key-file ./pd-tls/tls.key --ca-file ./pd-tls/ca.crt member list
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:33]: 这样 cert 的路径直接用绝对路径，方便 copy？

File: zh/enable-tls-between-components.md (hunk 5 above)
Comment (HuSharp) [reply zh/enable-tls-between-components.md:30]: 感觉放到 pod 里面会更加方便一点？

File: zh/enable-tls-between-components.md (hunk 6 above)
Comment (csuzhangxc) [reply zh/enable-tls-between-components.md:33]: 另外，我下载的 etcdctl 3.4.33，似乎这些证书文件的参数名不完全一样

File: zh/enable-tls-between-components.md [hunk 7]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    >    > 输出示例：
+    >    > 8e9e05c52164694d: name=tidb-test-v75-pd-0 peerURLs=http://localhost:2380 clientURLs=https://localhost:2379 isLeader=true
+    >    > ```
+    >    >
+    > 5. 修改 etcd member 的 peerURLs 为 https：
+    >    >
+    >    > ```bash
+    >    > etcdctl --endpoints https://127.0.0.1:2379 --cert-file ./pd-tls/tls.crt --key-file ./pd-tls/tls.key --ca-file ./pd-tls/ca.crt member update 8e9e05c52164694d --peer-urls="https://localhost:2380"
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:41]: memberID 和 URL 强调说明一下，免得复制粘贴时搞成 `localhost` 了？

File: zh/enable-tls-between-components.md [hunk 8]
```diff
@@ -21,6 +21,30 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+    >    > 输出示例：
+    >    > Updated member with ID 8e9e05c52164694d in cluster
+    >    > ```
+    >    >
+    > 6. 查看此时 etcd member 的 peerURLs 已经修改为 https。
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:46]: 再把 `member list` 的命令贴一遍？

File: zh/enable-tls-between-components.md (hunk 6 above)
Comment (HuSharp) [reply zh/enable-tls-between-components.md]: 已更新

File: zh/enable-tls-between-components.md [hunk 9]
```diff
@@ -1656,3 +1658,67 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
     cd /var/lib/cluster-client-tls
     /tikv-ctl --ca-path=ca.crt --cert-path=tls.crt --key-path=tls.key --host 127.0.0.1:20160 cluster
     ```
+
+## 从非 TLS 集群升级到 TLS 集群
+
```
Comment (HuSharp) [inline zh/enable-tls-between-components.md:1663]: 新开了一个标题，更新了相关描述，PTAL, thx~ @csuzhangxc

File: zh/enable-tls-between-components.md [hunk 10]
```diff
@@ -1656,3 +1658,71 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+> **注意：请谨慎操作**
+> 
+> 在集群无法重建的情况下，适用于集群创建后再开启 TLS。
+> 
+
+1. 如果存在多个 PD 节点，缩容 PD 至 1 个节点。
+
+2. 参考 [第一步：为 TiDB 集群各个组件生成证书](#第一步为-tidb-集群各个组件生成证书) ，准备证书及创建 Kubernetes Secret 对象。
+
+3. 参考 [第二步：部署 TiDB 集群](#第二步部署-tidb-集群) ，修改 `tidb-cluster.yaml` 文件，使用 `kubectl apply -f tidb-cluster.yaml` 来更新 TiDB 集群。
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:1673]: 这个 `3` 说明一下是  “更新” 什么？或者要不要准备一条可以复制粘贴的 `kubectl patch` 命令。

另外，“更新” 后应该要说明一下等待 PD Pod 重建？

Comment (HuSharp) [reply zh/enable-tls-between-components.md]: 加上了 patch 指令

File: zh/enable-tls-between-components.md [hunk 11]
```diff
@@ -1656,3 +1658,97 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
...
+
+    输出示例：
+
+    ``` shell
+    tidbcluster.pingcap.com/basic patched
+    ```
+
+    也可以参考 [第二步：部署 TiDB 集群](#第二步部署-tidb-集群)，选择设置 `cert-allowed-cn` 配置项（TiDB 为 `cluster-verify-cn`），用来验证集群间各组件证书的 CN (Common Name)。
+
+4. 登录 PD pod，下载 etcdctl。参考 [etcdctl 安装指南](https://etcd.io/docs/v3.4/install/)，etcdctl 位于解压后的文件夹目录下。
```
Comment (csuzhangxc) [inline zh/enable-tls-between-components.md:1695]:
```suggestion
4. 使用 `kubectl exec` 进入 PD pod，下载 etcdctl。参考 [etcdctl 安装指南](https://etcd.io/docs/v3.4/install/)，etcdctl 位于解压后的文件夹目录下。
```

**Review by csuzhangxc — APPROVED** [review]

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by mayjiang0203 — COMMENTED** [review]
shall we add docs for how to disable tls in tls enabled cluster ?

**Comment by csuzhangxc** [comment]
> shall we add docs for how to disable tls in tls enabled cluster ?

We need to do more tests, it may not work. ref https://github.com/pingcap/tidb-operator/issues/5728

**Comment by Oreoxmt** [comment]
@HuSharp I've submitted a PR that revises the wording and structure of this document, PTAL https://github.com/HuSharp/docs-tidb-operator/pull/1

**Comment by HuSharp** [comment]
> @HuSharp I've submitted a PR that revises the wording and structure of this document, PTAL [HuSharp#1](https://github.com/HuSharp/docs-tidb-operator/pull/1)

I have sent you my personal [repository](https://github.com/HuSharp/docs-tidb-operator) permissions invitation, feel free to change it :-)

**Review by Oreoxmt — APPROVED** [review]

File: zh/enable-tls-between-components.md [hunk 12]
```diff
@@ -20,7 +20,8 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/enable-tls-between-components/']
 
     > **注意：**
     >
-    > 在集群创建后，不能修改此字段，否则将导致集群升级失败，此时需要删除已有集群，并重新创建。
+    > - 在集群创建后，不能修改此字段，否则将导致集群升级失败，此时需要删除已有集群，并重新创建。
+    > - 若集群无法重建且需要启用 TLS，请参阅[从非 TLS 集群升级到 TLS 集群](#从非-tls-集群升级到-tls-集群)。
```
Comment (Oreoxmt) [inline zh/enable-tls-between-components.md:24]:
```suggestion
    > - 若集群无法重建且需要启用 TLS，请参阅[从非 TLS 集群升级到 TLS 集群](#将非-tls-集群升级为-tls-集群)。
```

**Pull request merged**